drracket/scribblings/tools/lang-tools.scrbl

```diff
@@ -74,98 +78,125 @@ The recognized token styles (specified implicitly via @method[color:text<%> star
           @item{@indexed-racket['parenthesis]}
           @item{@indexed-racket['error]}
           @item{@indexed-racket['other]}]
-These precise colors for these identifiers are controlled by the preferences dialog in DrRacket.
+
+The precise colors for these identifiers are controlled by the
+preferences dialog in DrRacket, and by other customization mechanisms
+in other tools.
+}
 
 @section{Indentation}
 
 @language-info-def[drracket:indentation]{
- When a language's @racket[_get-info] procedure responds to @racket['drracket:indentation],
- it is expected to return a function with this contract:
- @racketblock[(-> (is-a?/c racket:text<%>)
-                  exact-nonnegative-integer?
-                  (or/c #f exact-nonnegative-integer?))]
- The function is used
- to indent lines in DrRacket. It is called with the position containing the line to be
- indented. It is expected to return the number of spaces that should appear at the beginning
- of the line or @racket[#f]. If @racket[#f] is returned,
- DrRacket uses the standard s-expression indentation rules.
+
+When a language's @racket[_get-info] procedure responds to
+@racket['drracket:indentation], it is expected to return a function
+with this contract:
+
+@racketblock[(-> (is-a?/c (or/c racket:text<%> color-textoid<%>))
```

Comment:
In this commit I changed `racket:text<%>` to `(or/c racket:text<%> color-textoid<%>)` and added a margin-note just below explaining.

Part of me worries, though, that someone will read this quickly, skip the margin note, and just use `racket:text<%>`.

Instead, I'd like the "read-quickly default" to be `color-textoid<%>`. IOW if people are in a rush, I'd them to do something for more compatibility -- expeditor, Racket Mode, and more, as well as DrRacket. (`racket:text<%>` should be just for backward compatibility for old langs. Or maybe a new lang if they have no choice (if that's even a scenario, I'm not sure?)).

So what do you think about just saying `color-textoid<%>` here, no `or/c`, and leave discussion of `racket:text<%>` to the margin-note?  (Do you see what I mean? If I'm not explaining well let me know and I'll try again. :smile:)

Reply:
I like your "actually" idea, will try some prose like that and push a new commit.

I think it will be OK if updates are lazy not eager -- OK if some old langs still use full `text%`, and get updated if/as/when someone cares? Someday the lang author does. Or maybe some user of it, who wants it to work also in expeditor or emacs, makes a PR. Or no one does, but that's OK cause no one's really using that lang anymore, or at least not for a long time. Meanwhile it still works in DrRacket.

Anyway that's my starting feeling. Mainly I want new langs, or older langs with new tool support added, to get off on the right foot with `color-textoid<%>`.

Reply:
> I think it will be OK if updates are lazy not eager -- OK if some old langs still use full `text%`, and get updated if/as/when someone cares? Someday the lang author does. Or maybe some user of it, who wants it to work also in expeditor or emacs, makes a PR. Or no one does, but that's OK cause no one's really using that lang anymore, or at least not for a long time. Meanwhile it still works in DrRacket.
> 
> Anyway that's my starting feeling. Mainly I want new langs, or older langs with new tool support added, to get off on the right foot with `color-textoid<%>`.

I'm okay with that. One niggling worry is that `color-textoid<%>` is missing some important method and it might put people off from this path since they'd just keep using `text%`. We would be able to flush out those kinds of problems earlier if we tried to convert stuff. That said, this is not a today problem! 😄